sway-lib-std/src/storage/storage_vec.sw

```diff
@@ -755,10 +783,131 @@ impl<V> StorageKey<StorageVec<V>> {
+    ///     let mut vec = Vec::<u64>::new();
+    ///     vec.push(5);
+    ///     vec.push(10);
+    ///     vec.push(15);
+    ///
+    ///     storage.vec.store_vec(vec);
+    ///
+    ///     assert(5 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    ///     assert(15 == storage.vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(write)]
+    pub fn store_vec(self, vec: Vec<V>) {
+        let slice = vec.as_raw_slice();
+        // Get the number of storage slots needed based on the size of bytes.
+        let number_of_bytes = slice.number_of_bytes();
+        let number_of_slots = (number_of_bytes + 31) >> 5;
+        let mut ptr = slice.ptr();
+
+        // The capacity needs to be a multiple of 32 bytes so we can 
+        // make the 'quad' storage instruction store without accessing unallocated heap memory.
+        ptr = realloc_bytes(ptr, number_of_bytes, number_of_slots * 32);
+
+        // Store `number_of_slots * 32` bytes starting at storage slot `key`.
+        let _ = __state_store_quad(sha256(self.field_id), ptr, number_of_slots);
+
+        // Store the length length, NOT the bytes. 
+        // This differs from the existing `write_slice()` function to be compatible with `StorageVec`.
+        write::<u64>(self.field_id, 0, number_of_bytes / __size_of::<V>());
+    }
+
+    /// Load a `Vec` from the `StorageVec`.
+    ///
+    /// # Returns
+    /// 
+    /// * [Option<Vec<V>>] - The vector constructed from storage or `None`.
+    ///
+    /// # Number of Storage Accesses
+    ///
+    /// * Reads - `2`
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     let mut vec = Vec::<u64>::new();
+    ///     vec.push(5);
+    ///     vec.push(10);
+    ///     vec.push(15);
+    ///
+    ///     storage.vec.store_vec(vec);
+    ///     let returned_vec = storage.vec.load_vec().unwrap();
+    ///
+    ///     assert(5 == returned_vec.get(0).unwrap());
+    ///     assert(10 == returned_vec.get(1).unwrap());
+    ///     assert(15 == returned_vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(read)]
+    pub fn load_vec(self) -> Option<Vec<V>> {
+        // Get the length of the slice that is stored.
+        match read::<u64>(self.field_id, 0).unwrap_or(0) {
+            0 => None,
+            len => {
+                // Get the number of storage slots needed based on the size.
+                let number_of_slots = ((len *  __size_of::<V>()) + 31) >> 5;
+                let ptr = alloc_bytes(number_of_slots * 32);
+                // Load the stored slice into the pointer.
+                let _ = __state_load_quad(sha256(self.field_id), ptr, number_of_slots);
+                Some(Vec::from(asm(ptr: (ptr, len *  __size_of::<V>())) { ptr: raw_slice }))
+            }
+        }
+    }
+}
+
+fn offset_calculator<T>(offset: u64) -> u64 {
```

Comment:
Should this be moved to [storage_api.sw](https://github.com/FuelLabs/sway/blob/bitzoic-vec-storage/sway-lib-std/src/storage/storage_api.sw)? Or is it only relevant here?

Reply:
This is a private function that is only relevant here